### PR TITLE
refactor(placement): introduce PlacementContext + mutate on pathname change

### PR DESCRIPTION
**Pull request by fiji-flo**
## Summary

Previously the logic for fetching lived within the component. By refactoring this we start to enable more complex queries for future enhancements.

Resolves MP-268.

---

## Screenshots

Nothing changes visually.

### CODE REVIEW

File: client/src/placement-context.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+export type PlacementData = PlacementStatus | PlacementError;
+
+const PLACEMENT_TYPES = ["docs", "search"];
+
+function hasPlacement(pathname: string): boolean {
+  const [, , typ] = pathname.split("/");
+  return PLACEMENT_TYPES.includes(typ?.toLowerCase?.());
+}
```
Comment (fiji-flo) [inline client/src/placement-context.tsx:42]: Not sure if I'm happy with that. But I couldn't see a better way. Eventually we should have a single request in this central place, but it depends on the page type with what parameters we query the api.

Comment (caugner) [reply client/src/placement-context.tsx:42]: I would tend to prefer a RegExp here: `/\/[^\/]+\/(docs|search)\//i`

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM, just three nits.

File: client/src/placement-context.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        throw Error(response.statusText);
+      }
+    },
+    {
+      revalidateIfStale: true,
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+    }
+  );
+  useEffect(() => {
```
Comment (caugner) [inline client/src/placement-context.tsx:92]:
```suggestion

  useEffect(() => {
```

File: client/src/placement-context.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  success = "success",
+  geoUnsupported = "geo_unsupported",
+  capReached = "cap_reached",
+}
+
+export interface Fallback {
+  click: string;
+  view: string;
+  copy: string;
+  image: string;
```
Comment (caugner) [inline client/src/placement-context.tsx:18]: Just checking: `copy` and `image` here are not optional, but they are in `PlacementStatus`?

**Review by caugner — APPROVED** [review]
LGTM and tested, just one additional separating line would be nice.

File: client/src/placement-context.tsx (hunk 3 above)
Comment (fiji-flo) [reply client/src/placement-context.tsx:18]: yes

**Pull request merged**